openpgp/packet/private_key.go

```diff
@@ -286,10 +291,10 @@ func (pk *PrivateKey) Serialize(w io.Writer) (err error) {
 			if err != nil {
 				return err
 			}
-			l = buf.Len()
+			priv, l = buf.Bytes(), buf.Len()
 			if pk.sha1Checksum {
 				h := sha1.New()
-				io.Copy(h, buf)
+				h.Write(priv)
```

Comment:
```suggestion
				h.Write(buf.Bytes())
```

Reply:
I agree, it's cleaner not to use `priv` for multiple purposes. Thanks.